src/plugins/data/common/search/utils.ts

```diff
@@ -11,17 +11,36 @@ import { AggTypesDependencies } from '..';
 import type { IKibanaSearchResponse } from './types';
 
 /**
+ * From https://github.com/elastic/elasticsearch/issues/55572: "When is_running is false, the query has stopped, which
+ * may happen due to ... the search failed, in which case is_partial is set to true to indicate that any results that
+ * may be included in the search response come only from a subset of the shards that the query should have hit."
  * @returns true if response had an error while executing in ES
  */
 export const isErrorResponse = (response?: IKibanaSearchResponse) => {
-  return !response || !response.rawResponse || (!response.isRunning && !!response.isPartial);
+  return (
+    !response ||
+    !response.rawResponse ||
+    (!response.isRunning &&
+      !!response.isPartial &&
+      // See https://github.com/elastic/elasticsearch/pull/97731. For CCS with ccs_minimize_roundtrips=true, isPartial
+      // is true if the search is complete but there are shard failures. In that case, the _clusters.details section
+      // will have information about those failures. This will also likely be the behavior of CCS with
+      // ccs_minimize_roundtrips=false and non-CCS after https://github.com/elastic/elasticsearch/issues/98913 is
+      // resolved.
+      !response.rawResponse?._clusters?.details)
```

Comment:
I left the same comment [here](https://github.com/elastic/kibana/pull/166544#discussion_r1330142659), sorry for the duplicate comment.

I am not sure that checking `_clusters.details` is enough. I think we must check _clusters.details.**clusterAlias**.failures
The new response for CCS is now including `_clusters.details` both for error and non error responses.
An example of non error `details` section is: 
```
"details": {
  "(local)": {
    "status": "successful",
    "indices": foo",
    "took": 123,
    "timed_out": false,
    "_shards": {
      ...
    }
  },
  "remote": {
    "status": "successful",
    "indices": "foo",
    "took": 234,
    "timed_out": false,
    "_shards": {
      ...
    }
  }
}
```
And an example of error `details` section:
```
"details": {
  "remote": {
      "status": "partial",
      "indices": "foo",
      "took": 123,
      "timed_out": false,
      "_shards": {
        "total": 6,
        "successful": 5,
        "skipped": 0,
        "failed": 1  
      },
      "failures": [ 
        {
          "shard": 1,
          "index": "remote:foo",
          "node": ...,
          "reason": {
            "type": "query_shard_exception",
            "reason": "failed to create query: exception message here",
            ...
          }
        }
      ]
  }
}
```

Reply:
Thanks for weighing in here @piergm. 

To give some context, this check (`isErrorResponse`) will soon be removed entirely (see https://github.com/elastic/kibana/pull/164506). Right now, if `isErrorResponse` returns `true`, then we throw an error from the Kibana server and clients will receive a 500 error. This behavior is not ideal, but it has been this way for years now and we were hoping to limit the scope as much as possible to speed up a potential patch release.

The change in this PR is less about identifying when a response from ES is/is not an actual error, but instead opening an escape hatch to **not** throw an error if we encounter the new response format. In other words, the check `!response.rawResponse?._clusters?.details` is not actually for identifying whether or not the response contains errors, but rather for identifying that the response format is the new format.

Reply:
Thank you for clarifying this @lukasolson. Than this LGTM!